inst/tests/tests.Rraw

```diff
@@ -18454,3 +18454,10 @@ test(2256.6, fread('a;b\n1,14;5', verbose=TRUE), data.table(a=1.14, b=5L), outpu
 # helpful error about deleting during grouping, #1873
 DT = data.table(id = c(1, 1, 2, 2), a = 1:4, b = 5:8)
 test(2257, DT[ , c("c", "a") := .(a + 1, NULL), by=id], error="it's not possible to delete parts of a column")
+
+DT = data.table(x=c(NA, "a", "b"))
+test(2258.1, capture.output(print(DT, na.print=".")), c("   x", "1: .", "2: a", "3: b"))
```

Comment:
We should try and get coverage for every call site, here I think we only cover one.

The remaining 3 require:

 - `col.names="none"`
 - `printdots`, with/without `col.names="none"`

Reply:
Done! PS: I'm assuming it is expected that `na.print` doesn't work with non-character datatables, as this is consistent with the behavior for printing dataframes as well:
```R
DF = data.frame(x = c(NA, 1))
print(DF, na.print=".")
#    x
# 1 NA
# 2  1
DF = data.frame(x = c(NA,"e"))
print(DF, na.print=".")
#   x
# 1 .
# 2 1
```
I originally wrote a test that was a combination of NAs and integers, and I was wondering for a very long time why it wasn't passing until I realized that we probably wouldn't want strings in our numerical datatables anyway :sob:.

Reply:
see also

https://bugs.r-project.org/show_bug.cgi?id=18701

and

https://bugs.r-project.org/show_bug.cgi?id=18702

inspired by this PR :)